src/compiler/mir/ir.rs

```diff
@@ -99,22 +109,29 @@ impl Display for Procedure {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         f.write_fmt(format_args!("fn () -> {:?}:\n", self.ret_ty))?;
 
+        // Print out the stack: vars and temps
+        f.write_str("Stack:\n")?;
+
         // Print out the variables
         for idx in 0..self.vars.len() {
+            let vid = VarId::new(idx);
             if self.vars[idx].mutable {
-                f.write_fmt(format_args!("let mut %{}: {:?} // {}\n", idx, self.vars[idx].ty, self.vars[idx].name))?
+                f.write_fmt(format_args!("let mut {}: {:?} // {}\n", vid, self.vars[idx].ty, self.vars[idx].name))?
```

Comment:
Add refs to the `StringTable` then this would be able to print the actual string values.  Or change this to CompilerDisplay